executor/read/pattern_executor.rs

```diff
@@ -62,18 +58,30 @@ impl PatternExecutor {
         self.control_stack.is_empty()
     }
 
-    pub(crate) fn compute_next_batch(
+    pub(crate) fn compute_next_batch_restore_and_retry_as_needed(
```

Comment:
I'd just call this `compute_next_batch_with_retries` "as needed" is a funny prefix for a method name :D